packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -26,14 +26,44 @@ typedef PointerExitEventListener = void Function(PointerExitEvent event);
 /// Used by [MouseTrackerAnnotation], [MouseRegion] and [RenderMouseRegion].
 typedef PointerHoverEventListener = void Function(PointerHoverEvent event);
 
+// A key, so that (_IdenticalAnnotationKey(a) == _IdenticalAnnotationKey(b)) is equivalent to
+// identical(a, b).
+class _IdenticalAnnotationKey implements Key {
+  _IdenticalAnnotationKey(this.value);
+
+  final MouseTrackerAnnotation value;
+
+  // Don't return `identityHashCode(value)` directly since the hash of this key
+  // needs to be different from that of `value`. Do a fast conversion with a
+  // random mask, since we don't need a secure hashing algorithm for a private
+  // class.
+  @override
+  int get hashCode => _hashCodeMask ^ identityHashCode(value);
+  static const int _hashCodeMask = 0xc366d8df10ee8;
```

Comment:
why does it need to be different? would we ever put instances of Key and instances of MouseTrackerAnnotation into the same map?

Reply:
No. If it's not required that hashCode must be different, I'm glad to remove the mask.